app/waf/src/controllers/wafpolicy.controller.ts

```diff
@@ -120,6 +129,45 @@ export class WafpolicyController extends BaseController {
     );
   }
 
+  @post(prefix + '/wafpolicies/{id}/ads/{adcId}', {
+    responses: {
+      '204': Schema.emptyResponse('Uploading WAF Policy resource'),
+      '404': Schema.notFound('Can not find trustedDeivceId'),
+    },
+  })
+  async wafpolicyUpload(
+    @param(Schema.pathParameter('id', 'WAF Policy resource ID')) id: string,
+    @param(Schema.pathParameter('adcId', 'ADC resource ID')) adcId: string,
+  ): Promise<void> {
+    let wafpolicy: Wafpolicy = await this.wafpolicyRepository.findById(id, {
+      where: {
+        and: [
+          {or: [{tenantId: await this.tenantId}, {public: true}]},
+          {id: id},
+        ],
+      },
+    });
+
+    let adc: Adc = await this.adcRepository.findById(adcId, undefined, {
+      tenantId: await this.tenantId,
+    });
+
+    if (adc.trustedDeviceId == undefined) {
+      throw new HttpErrors.NotFound(
```

Comment:
I feel it might be better to return 422 or 400, instead of 404. What do you think?